Split the function getCatalog in Product

JethroWarnett opened the pull request:
The function getCatalog in Product was split to getEntireCatalog and getAvailableCatalog. The latter returns all products and the former returns all products that are available

JethroWarnett committed:
Split the function getCatalog in Product

The function getCatalog in Product was split to getEntireCatalog and getAvailableCatalog. The latter returns all products and the former returns all products that are available

JethroWarnett committed:
Fixed some errors

There were errors in the previous commit. These were fixed. The code seems to work with Postman

JethroWarnett committed:
Merge branch 'master' into AvailableProductsCatalog

dominiqueroux commented on backend/src/services/product.service.ts:
```diff
@@ -85,8 +85,20 @@ export class ProductService {
+    public getAvailableCatalog(): Promise<Product[]> {
+        return Product.findAll({
+            where: {
+                [Op.or]: [
+                    {status: 1},
+                    {approved: 1}]
+            }
```
Fails here
you need to import this

JethroWarnett committed:
Installed multer and importet Op to fix crash

The code was crashing beforehand. Once I installed multer and importet Op from sequelize, the problem stopped

dominiqueroux committed:
took the db.sqlite from master

JethroWarnett committed:
Merge branch 'master' into AvailableProductsCatalog

dominiqueroux committed:
Merge branch 'master' into AvailableProductsCatalog
